docs/data/base/guides/overriding-component-structure/overriding-component-structure.md

```diff
@@ -0,0 +1,113 @@
+
+{{"demo": "OverridingRootSlot.js"}}
+
+:::success
+If you provide a non-interactive element like a `<div>` or a `<span>`, the Button will automatically add the necessary accessibility attributes.
+Try inspecting the demo Button above in your browser's dev tools to see this feature in action.
+:::
+
+## Interior slots
+
+Complex components are composed of one or more interior slots in addition to the root.
+These slots are often (but not necessarily) nested within the root.
+
+For example, the [Unstyled Slider](/base/react-slider/) is composed of a root `<span>` that houses several interior slots named for the elements they represent: track, thumb, rail, and so on.
+
+### The slots prop
+
+Use the `slots` prop to override a component's interior slots.
+The example below shows how to override the listbox slot in the [Unstyled Select](/base/react-select/) component—a `<ul>` by default—with an `<ol>`:
+
+{{"demo": "OverridingInternalSlot.js"}}
+
+Note that you can also use the `slots` prop to override the root slot:
+
+```jsx
+// This:
+<SelectUnstyled slots={{ root: 'span' }} />
+
+// ...is the same as this:
+<SelectUnstyled component="span">
+```
+
+But if you try to override the root slot with both `component` and `slots`, then `component` will take precedence:
+
+```jsx
+// This:
+<SelectUnstyled component="div" slots={{ root: 'span' }} />
+
+// ...renders as this:
+<div class="MuiSelectUnstyled-root" />
+```
+
+### The slotProps prop
+
+The `slotProps` prop is an object that contains the props for all slots within a component.
+You can use it to define additional custom props to pass to a component's interior slots.
+
+For example, the code snippet below shows how to add a custom CSS class to the badge slot of the [Unstyled Badge](/base/react-badge/) component:
+
+```jsx
+<BadgeUnstyled slotProps={{ badge: { className: 'my-badge' } }} />
+```
+
+All additional props placed on the primary component are also propagated into the root slot (just as if they were placed in `slotProps.root`).
+These two examples are equivalent:
+
+```jsx
+<BadgeUnstyled id="badge1">
+```
+
+```jsx
+<BadgeUnstyled slotProps={{ root: { id: 'badge1' } }}>
+```
+
+:::warning
+If both `slotProps.root` and additional props have the same keys but different values, the `slotProps.root` props will take precedence.
+This does not apply to classes or the `style` prop—they will be merged instead.
+:::
+
+## Best practices
+
+If you are customizing a simpler component like the Unstyled Button that only has a root slot, you may prefer to use the more succinct `component` prop instead of `slots`.
+
+Overriding with `component` lets you apply the attributes of that element directly to the root.
+For instance, if you replace the Unstyled Button root with an `<li>` tag, you can add the `<li>` attribute `value` directly to the component.
+If you did the same with `slots.root`, you would need to place this attribute on the `slotProps.root` object in order to avoid a TypeScript error.
```

Comment:
```suggestion
```

I think this could be deleted for Joy docs, otherwise, it could confuse the reader.